### PR TITLE
Fix data leakage in preprocessing

### LINKED ISSUE #1574: [Bug] Data leakage in Automated Financial Reporting — scaler fitted on full dataset before split
## 🐛 Bug Report

**File:** `Automated Financial Reporting with Deep Learning/app.py` — Line 16

**Description:**
The `MinMaxScaler` is fitted on the **entire dataset** before any train/test split. This
is a classic **data leakage** mistake — test set statistics leak into the scaler, making
model accuracy appear better than it really is. Since this is a learning repo, it actively
teaches beginners a wrong ML practice.

## Current Code (Buggy)
```python
scaler = MinMaxScaler()
data_scaled = scaler.fit_transform(df.drop('Date', axis=1))  # leaks test stats ✗
```

## Fix
```python
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

scaler = MinMaxScaler()
X_train_scaled = scaler.fit_transform(X_train)  # fit ONLY on train ✓
X_test_scaled = scaler.transform(X_test)         # only transform test ✓
```

**Severity:** 🔴 High — teaches incorrect ML methodology to beginners

### PR DESCRIPTION
**Pull request by chandni5033**
## Description
Fixed data leakage issue caused by applying MinMaxScaler before train-test split.

## Changes Made
- Added train_test_split
- Applied fit_transform only on training data
- Applied transform on test data

Fixes #1574

### CODE REVIEW

**Review by Sibam-Paul — APPROVED** [review]
looks good
